frontend/src/Hooks/useUserAccess.js

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from 'react'
+import { loadFromLocal, saveToLocal } from '../Lib/localStorage'
+import { loginUser, signUpUser } from '../Services/handleApiUserAccess'
+
+export default function useUser() {
+  const [userData, setUserData] = useState()
+  const notAuthorized = { authorized: false }
```

Comment:
In your App.js you call it "unauthorized". Maybe it's better to name it consistently. Or is there a reason to make it different?